src/loader.ts

```diff
@@ -514,19 +514,35 @@ const globalObject: any = Function('return this')();
 		let module: DojoLoader.Module;
 		const pluginRegEx = /^(.+?)\!(.*)$/;
 
-		// Update moduleId from map if exists
-		moduleId = updateModuleIdFromMap(moduleId, referenceModule);
-
-		// check if module is a plugin-module
-		const match = moduleId.match(pluginRegEx);
+		// Foreseable situations (where ?-> is a map lookup function)
+		// module
+		// plugin!arg
+		// module ?-> mappedModule
+		// module ?-> mappedPlugin!arg
+		// plugin!arg ?-> mappedPlugin + ! + arg
+
+		// Do inital check on the passed in moduleId
+		let passedModuleMatch = moduleId.match(pluginRegEx);
```

Comment:
I know you were just adjusting what was there, but this never gets re-assigned, therefore should be a `const`.